Firestore/tests/System/FirestoreMultipleDbTest.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Firestore\Tests\System;
+
+use Google\Cloud\Firestore\FieldPath;
+use Google\Cloud\Firestore\Query;
+
+/**
+ * @group firestore
+ * @group firestore-multipledb
+ */
+class FirestoreMultipleDbTest extends FirestoreTestCase
+{
+    private $document;
+
+    public function setUp(): void
+    {
+        $this->document = self::$multiDbCollection->newDocument();
+    }
+
+    public function testInsert()
+    {
+        $this->assertFalse($this->document->snapshot()->exists());
+
+        self::$multiDbClient->runTransaction(function ($t) {
+            $t->create($this->document, [
+                'foo' => 'bar'
+            ]);
+        });
+
+        $this->assertTrue($this->document->snapshot()->exists());
+    }
+
+    public function testUpdate()
+    {
+        $this->document->create([
+            'foo' => 'bar'
+        ]);
+
+        self::$multiDbClient->runTransaction(function ($t) {
+            $t->update($this->document, [
+                ['path' => 'bat', 'value' => 'baz']
+            ]);
+        });
+
+        $this->assertEquals([
+            'foo' => 'bar',
+            'bat' => 'baz'
+        ], $this->document->snapshot()->data());
+    }
+
+
+    public function testCollectionGroup()
+    {
+        list ($query) = $this->createDocuments([
+            'abc/123/%s/cg-doc1',
+            'abc/123/%s/cg-doc2',
+            '%s/cg-doc3',
+            '%s/cg-doc4',
+            'def/456/%s/cg-doc5',
+            '%s/virtual-doc/nested-coll/not-cg-doc',
+            'x%s/not-cg-doc',
+            '%sx/not-cg-doc',
+            'abc/123/%sx/not-cg-doc',
+            'abc/123/x%s/not-cg-doc',
+            'abc/%s',
+        ]);
+
+        // Returns docs with parent path having the collection id
```

Comment:
I am still confused... why doesn't `not-cg-doc` show up when getting the query IDs?

I still think it's hard to follow what's going on in this test.

Reply:
None of the `*/not-cg-doc` are having the `/collection-group-name/`. Its prefixed by `x`, so they wont appear in the query.

Reply:
I'm more asking that you construct the tests and add comments in a way that explains what's going on, not that you explain it to me in this comment thread. For the next person who sees your code.

With that being said, I'm still confused. First, nothing in the query says (afaict) that they can't be prefixed with an x. Secondly, several `not-cg-doc` entries are NOT prefixed with an `x` (such as `%s/virtual-doc/nested-coll/not-cg-doc`, `abc/%s`, and several which are _suffixed_ with the `x`).

Something like this would suffice:

```php
            // these paths will match based on the collection group
            'abc/123/%s/cg-doc1',
            'abc/123/%s/cg-doc2',
            '%s/cg-doc3',
            '%s/cg-doc4',
            'def/456/%s/cg-doc5',

            // these paths will not match
            '%s/virtual-doc/nested-coll/not-cg-doc', // nested column
            'x%s/not-cg-doc',                        // x-prefix
            '%sx/not-cg-doc',                        // x-suffix
            'abc/123/%sx/not-cg-doc',                // x-prefix
            'abc/123/x%s/not-cg-doc',                // x-suffix
            'abc/%s',                                // collection group as ID

```

Reply:
fixed